skipchain/api.go

```diff
@@ -135,9 +158,56 @@ func (c *Client) CreateGenesisSignature(ro *onet.Roster, baseH, maxH int, ver []
 	if err != nil {
 		return nil, err
 	}
+
+	if testCorruptSSBResponse != nil {
+		sb = testCorruptSSBResponse
+	}
+
+	// at this point we only know that the hash is correct but we need to compare
+	// what is inside the block (e.g. roster, verifiers, ...) because the distant
+	// node could have changed that
+	err = compareGenesisBlocks(genesis, sb.Latest)
+	if err != nil {
+		return nil, err
+	}
+
 	return sb.Latest, nil
 }
 
+// CompareGenesisBlocks compares the content of two blocks and returns an error
+// if there is any difference
+func compareGenesisBlocks(prop *SkipBlock, ret *SkipBlock) error {
```

Comment:
why can't you re-calculate the hash? if the hash is different then the block content should also be different right?

Reply:
I wish I could but the service is adding information to the genesis like a random backlink